website/search/elastic_search.py

```diff
@@ -578,6 +625,52 @@ def update_institution(institution, index=None):
 
         client().index(index=index, doc_type='institution', body=institution_doc, id=id_, refresh=True)
 
+
+@celery_app.task(bind=True, max_retries=5, default_retry_delay=60)
+def update_cgm_async(self, cgm_id, collection_id=None, op='update', index=None):
+    CollectedGuidMetadata = apps.get_model('osf.CollectedGuidMetadata')
+    if collection_id:
+        try:
+            cgm = CollectedGuidMetadata.objects.filter(
+                guid___id=cgm_id,
+                collection_id=collection_id,
+                collection__provider__isnull=False,
+                collection__deleted__isnull=True,
+                collection__is_bookmark_collection=False).first()
```

Comment:
`first` will return `None` if there aren't any -- the `except` here will never be hit. Likewise below

Reply:
ah that was left off in debugging some failing tests where cgm was not found because the collection was non-provided.